src/verso/Verso/Doc.lean

```diff
@@ -631,32 +631,35 @@ private partial def Part.reprPrec [Repr genre.Inline] [Repr genre.Block] [Repr g
 public instance [Repr g.Inline] [Repr g.Block] [Repr g.PartMetadata] : Repr (Part g) where
   reprPrec := private Part.reprPrec
 
+public structure DocReconstruction where
+
 /--
 The result type of values created by Verso's `#doc` and `#docs` commands. A value of type
 {lean}`VersoDoc` represents a not-fully-evaluated document of type {lean}`Part` that can be turned
 into a value by invoking the `VersoDoc.toPart` method. The actual structure of a {lean}`VersoDoc`
 should not be relied on.
 -/
 public structure VersoDoc (genre : Genre) where
-  construct : Unit → Part genre
+  construct : DocReconstruction → Part genre
```

Comment:
Presumably this will be:
```suggestion
  construct : genre.DocReconstruction → Part genre
```
in the end, right? And then `DocElabM` will have one of these in its state?